rl4co/models/nn/improvement_attention.py

```diff
@@ -0,0 +1,234 @@
+from typing import Callable, Optional, Tuple, Union
+
+import torch
+import torch.nn as nn
+from torch import Tensor
+import torch.nn.functional as F
+import math
+
+from rl4co.models.nn.attention import MultiHeadAttention
+from rl4co.models.nn.ops import Normalization, SkipConnection
+from rl4co.utils.pylogger import get_pylogger
+
+log = get_pylogger(__name__)
+
+class mySequential(nn.Sequential):
+
+    def forward(
+        self, *inputs: Union[Tuple[torch.Tensor], torch.Tensor]
+    ) -> Union[Tuple[torch.Tensor], torch.Tensor]:
+        for module in self._modules.values():
+            if type(inputs) == tuple:
+                inputs = module(*inputs)
+            else:
+                inputs = module(inputs)
+        return inputs
+
+
+class MultiHeadCompat(nn.Module):
+    def __init__(
+            self,
+            n_heads,
+            input_dim,
+            embed_dim=None,
+            val_dim=None,
+            key_dim=None
+    ):
+        super(MultiHeadCompat, self).__init__()
+    
+        if val_dim is None:
+            # assert embed_dim is not None, "Provide either embed_dim or val_dim"
+            val_dim = embed_dim // n_heads
+        if key_dim is None:
+            key_dim = val_dim
+
+        self.n_heads = n_heads
+        self.input_dim = input_dim
+        self.embed_dim = embed_dim
+        self.val_dim = val_dim
+        self.key_dim = key_dim
+
+        self.W_query = nn.Parameter(torch.Tensor(n_heads, input_dim, key_dim))
+        self.W_key = nn.Parameter(torch.Tensor(n_heads, input_dim, key_dim))
+
+        self.init_parameters()
+        
+    def init_parameters(self):
+
+        for param in self.parameters():
+            stdv = 1. / math.sqrt(param.size(-1))
+            param.data.uniform_(-stdv, stdv)
```

Comment:
[Minor] As @fedebotu mentioned and refer to torch initialize rule at [here](https://pytorch.org/docs/stable/generated/torch.nn.Linear.html#). We could clean up this~